src/forwardBeamPipeBrazil.cpp

```diff
@@ -0,0 +1,570 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Alex Jentsch
+
+//==========================================================================
+//
+//      <detector name ="DetName" type="Beampipe" >
+//      <layer id="#(int)" inner_r="#(double)" outer_z="#(double)" >
+//      <slice material="string" thickness="#(double)" >
+//      </layer>
+//      </detector>
+//==========================================================================
+#include "DD4hep/DetFactoryHelper.h"
+#include "DD4hep/Printout.h"
+#include "TMath.h"
+#include <XML/Helper.h>
+
+using namespace std;
+using namespace dd4hep;
+
+/** \addtogroup beamline Beamline Instrumentation
+ */
+
+/** \addtogroup IRChamber Interaction Region Vacuum Chamber.
+ * \brief Type: **IRChamber**.
+ * \ingroup beamline
+ *
+ *
+ * \code
+ *   <detector>
+ *   </detector>
+ * \endcode
+ *
+ */
+static Ref_t create_detector(Detector& det, xml_h e, SensitiveDetector /* sens */)
+{
+
+        using namespace ROOT::Math;
+        xml_det_t x_det    = e;
+        string    det_name = x_det.nameStr();
+        // Material   air       = det.air();
+        DetElement sdet(det_name, x_det.id());
+        Assembly   assembly(det_name + "_assembly");
+        // Material   m_Cu    = det.material("Copper");
+        Material m_Al     = det.material("Aluminum");
+        Material m_Be     = det.material("Beryllium");
+        Material m_SS     = det.material("StainlessSteel");
+        Material m_vac    = det.material("Vacuum");
+        string   vis_name = x_det.visStr();
+
+        PlacedVolume pv_assembly;
+
+        double b0_hadron_tube_inner_r = 2.9 * dd4hep::cm;
+        double b0_hadron_tube_outer_r = 3.1 * dd4hep::cm;
+        double b0_hadron_tube_length  = 120.0 * dd4hep::cm;
+
+        double pipeThickness     = 5.0;
+
+        Double_t length[10];
+        Double_t innerXRadius[10];
+        Double_t innerYRadius[10];
+        Double_t outerXRadius[10];
+        Double_t outerYRadius[10];
+        Double_t rotationAngle[10];
+        Double_t xCenter[10];
+        Double_t yCenter[10];
+        Double_t zCenter[10];
```

Comment:
Replace this with a std::vector of struct with all these fields, i.e.
```suggestion
        struct beampipe_t {
          Double_t length,
          Double_t innerXRadius,
          Double_t innerYRadius,
          Double_t outerXRadius,
          Double_t outerYRadius,
          Double_t rotationAngle,
          Double_t xCenter,
          Double_t yCenter,
          Double_t zCenter,
        };
        std::vector<beampipe_t> beampipes;
```
This results in having values for all necessary components all the time, and it allows for inserting beam pipes without then having to go through and updating all indices.